includes/class-wc-payments-token-service.php

```diff
@@ -120,9 +122,7 @@ public function woocommerce_get_customer_payment_tokens( $tokens, $user_id, $gat
 			$stored_tokens = [];
 
 			foreach ( $tokens as $token ) {
-				if ( WC_Payment_Gateway_WCPay::GATEWAY_ID === $token->get_gateway_id() ) {
-					$stored_tokens[ $token->get_token() ] = $token;
-				}
+				$stored_tokens[ $token->get_token() ] = $token;
```

Comment:
If I'm reading this right, this line previously checked to make sure that we were only grabbing WCPay gateway tokens and now we're displaying all tokens - could WCPay potentially try to display tokens from other gateways? You could test this with the default Stripe extension or something. 🤔 
Perhaps we could check here if the gateway is within the array of WCPay gateways.

Reply:
This didn't affect display of tokens but it did cause tokens created by other gateways to be deleted when `$stored_tokens` was compared with `$payment_methods`, since those tokens weren't payment methods in the express account. Thanks for catching this! Fixed in https://github.com/Automattic/woocommerce-payments/pull/4670/commits/cc5a465171d908dc4f12e41563d39a6d9ce32a04.